core-backend/sandbox/src/funcs.rs

```diff
@@ -120,11 +122,13 @@ impl<E: Ext + 'static> FuncsHandler<E> {
             let dest: ProgramId = funcs::get_bytes32(memory, program_id_ptr)?.into();
             let payload = funcs::get_vec(memory, payload_ptr, payload_len)?;
             let value = funcs::get_u128(memory, value_ptr)?;
-            ext.send(HandlePacket::new(dest, payload, value))
-                .map_err(FuncError::Core)
-                .on_success_code(|message_id| wto(memory, message_id_ptr, message_id.as_ref()))
+            Ok(ExtErrorProcessor::new(ext)
+                .with(|ext| ext.send(HandlePacket::new(dest, payload, value)))
+                .map_err(FuncError::Core)?
+                .on_success(|message_id| wto(memory, message_id_ptr, message_id.as_ref()))?
+                .error_len())
         })
-        .map(|code| Value::I32(code).into())
+        .map(|code| Value::I32(code as i32).into())
```

Comment:
why do we need type cast here? what's return type of inner closure?

Reply:
`error_len()` returns u32 and syscalls themselves wait for it but sandbox allows to pass only signed integers